test/unix/main.ml

```diff
@@ -143,23 +238,86 @@ let readonly () =
               (Printf.sprintf "Wrong insertion: %s value is missing." v))
     !l
 
+let readonly_no_flush () =
+  let w =
+    Index.v ~fresh:true ~readonly:false ~log_size ~fan_out_size index_name
+  in
+  let r =
+    Index.v ~shared:false ~fresh:false ~readonly:true ~log_size ~fan_out_size
+      index_name
+  in
+  List.iter (fun (k, v) -> Index.add w k v) !l;
+  test_find_present r
+
+let readonly_lwt () =
+  let w =
+    Index.v ~fresh:true ~readonly:false ~log_size ~fan_out_size index_name
+  in
+  let r =
+    Index.v ~shared:false ~fresh:false ~readonly:true ~log_size ~fan_out_size
+      index_name
+  in
+  let wp = Lwt_list.iter_p (fun (k, v) -> Lwt.return (Index.add w k v)) !l in
+  let p =
+    Lwt.bind
```

Comment:
I suggest using the infix synonym for bind (`>>=`), to let this read more sequentially. It's common to open `Lwt.Infix` and use those operators instead of the standard functions.